modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1146,6 +1182,13 @@ def _take_2d_positional(
 
             if self.has_materialized_dtypes:
                 new_dtypes = self.dtypes.iloc[monotonic_col_idx]
+            elif isinstance(self._dtypes, ModinDtypes):
+                try:
+                    new_dtypes = self._dtypes.lazy_get(
+                        monotonic_col_idx, numeric_index=True
+                    )
+                except NotImplementedError:
```

Comment:
Should we change it here as well. According to https://github.com/modin-project/modin/pull/6663#discussion_r1396273042?